core/app/models/spree/promotion_duplicator.rb

```diff
@@ -0,0 +1,30 @@
+class Spree::PromotionDuplicator
```

Comment:
That makes sense, especially with rules and actions handling their own duplication